src/component/administration/user/CreateNewUserForm.tsx

```diff
@@ -0,0 +1,272 @@
+        if (_.isBoolean(action)) {
+          setUsernameExists(action);
+        }
+      });
+    }, Constants.INPUT_DEBOUNCE_WAIT_TIME)
+  ).current;
+
+  const onFormSubmit = () => {
+    if (!isValid()) {
+      return;
+    }
+
+    onSubmit({
+      firstName,
+      lastName,
+      username,
+      password: emailPassword ? "" : password,
+    });
+  };
+
+  const onChange = (
+    setter: (value: any) => void,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setter(e.currentTarget.value);
+  };
+
+  const onKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter" && isValid()) {
+      onFormSubmit();
+    }
+  };
+
+  const validateNonEmpty = useRef((value: string) => {
+    if (value == null || value.trim().length === 0) {
+      return ValidationResult.BLOCKER;
+    }
+    return ValidationResult.VALID;
+  }).current;
+
+  const validateUsername = useRef(
+    (username: string, usernameExists?: boolean) => {
+      if (validateNonEmpty(username) === ValidationResult.BLOCKER) {
+        return ValidationResult.BLOCKER;
+      }
+      if (!Utils.isValidEmail(username)) {
+        return ValidationResult.blocker(
+          i18n("register.username.notValidEmail")
+        );
+      }
+      if (usernameExists) {
+        return ValidationResult.blocker(
+          i18n("register.username-exists.tooltip")
+        );
+      }
+
+      return ValidationResult.VALID;
+    }
+  ).current;
+
+  // on username change, contact API and check if username already exists
+  useEffect(() => {
+    if (validateUsername(username) === ValidationResult.VALID) {
+      fetchUsernameExists(username);
+    }
+  }, [username, validateUsername, fetchUsernameExists]);
+
+  const validatePasswordConfirmation = () => {
+    if (password !== passwordConfirmation) {
+      return ValidationResult.blocker(
+        i18n("register.passwords-not-matching.tooltip")
+      );
+    }
+    if (validateNonEmpty(passwordConfirmation) === ValidationResult.BLOCKER) {
+      return ValidationResult.BLOCKER;
+    }
+    return ValidationResult.VALID;
+  };
+
+  const onEnterPasswordToggle = () => {
+    setEmailPassword(!emailPassword);
+  };
+
+  const isValid = () => {
+    const validationResults = [
+      validateNonEmpty(firstName),
+      validateNonEmpty(lastName),
+      validateUsername(username, usernameExists),
+    ];
+    if (!emailPassword) {
+      validationResults.push(
+        validateNonEmpty(password),
+        validatePasswordConfirmation()
+      );
+    }
+
+    // if there is at least one "false" value, the form is invalid
+    return !validationResults
+      // map validation results to booleans
+      .map((result) => result.severity === Severity.VALID)
+      // search "false" value
+      .includes(false);
+  };
+
+  return (
+    <Container>
+      <PromiseTrackingMask area="CreateNewUserForm" />
+      <Form>
+        <Row>
+          <Col md={6}>
+            <EnhancedInput
```

Comment:
The inputs should be all small

Reply:
every input with `md={6}` ?

Reply:
Every input in the app. Can be probably handled by passing it in `EnhancedInput` which, in contrast to `CustomInput` does not do it.